src/components/scenes/SendScene2.tsx

```diff
@@ -0,0 +1,924 @@
+        }
+      }
+      addToFioAddressCache(account, payeeFioAddresses)
+
+      if (broadcastedTx.metadata == null) {
+        broadcastedTx.metadata = {}
+      }
+      broadcastedTx.metadata.name = payeeName
+
+      if (payeeName != null && fioSender != null) {
+        let fioNotes = `${s.strings.fragment_transaction_list_sent_prefix}${s.strings.fragment_send_from_label.toLowerCase()} ${fioSender.fioAddress}\n`
+        fioNotes += fioSender.memo ? `\n${s.strings.fio_sender_memo_label}: ${fioSender.memo}\n` : ''
+        if (notes.length > 1) {
+          fioNotes += notes.join('\n')
+        }
+        broadcastedTx.metadata.notes = `${fioNotes}\n${broadcastedTx.metadata?.notes ?? ''}`
+      }
+
+      const { name, type, id } = coreWallet
+      const {
+        currencyCode,
+        nativeAmount,
+        networkFee,
+        parentNetworkFee,
+        txid,
+        ourReceiveAddresses,
+        deviceDescription,
+        networkFeeOption,
+        requestedCustomFee,
+        feeRateUsed
+      } = signedTx
+
+      logActivity(`broadcastTx: ${account.username} -- ${name ?? 'noname'} ${type} ${id}`)
+      logActivity(`
+  currencyCode: ${currencyCode}
+  nativeAmount: ${nativeAmount}
+  txid: ${txid}
+  networkFee: ${networkFee}
+  parentNetworkFee: ${parentNetworkFee ?? ''}
+  deviceDescription: ${deviceDescription ?? ''}
+  networkFeeOption: ${networkFeeOption ?? ''}
+  requestedCustomFee: ${JSON.stringify(requestedCustomFee)}
+  feeRateUsed: ${JSON.stringify(feeRateUsed)}
+  spendTargets: ${JSON.stringify(spendInfo.spendTargets)}
+  ourReceiveAddresses: ${JSON.stringify(ourReceiveAddresses)}`)
+
+      await coreWallet.saveTx(broadcastedTx)
+
+      for (const target of spendInfo.spendTargets) {
+        // Write FIO OBT per spendTarget
+        await recordFioObtData(target, currencyCode, broadcastedTx.txid)
+      }
+
+      playSendSound().catch(error => console.log(error)) // Fail quietly
+
+      if (onDone) {
+        onDone(null, broadcastedTx)
+      } else {
+        Actions.replace('transactionDetails', {
+          edgeTransaction: broadcastedTx,
+          walletId
+        })
+      }
+      Alert.alert(s.strings.transaction_success, s.strings.transaction_success_message, [
+        {
+          onPress() {},
+          style: 'default',
+          text: s.strings.string_ok
+        }
+      ])
+    } catch (e: any) {
+      resetSlider()
+      console.log(e)
+      let message = sprintf(s.strings.transaction_failure_message, e.message)
+      e.message = 'broadcastError'
+      if (e.name === 'ErrorEosInsufficientCpu') {
+        message = s.strings.send_confirmation_eos_error_cpu
+      } else if (e.name === 'ErrorEosInsufficientNet') {
+        message = s.strings.send_confirmation_eos_error_net
+      } else if (e.name === 'ErrorEosInsufficientRam') {
+        message = s.strings.send_confirmation_eos_error_ram
+      } else if (e.code && e.code === FIO_NO_BUNDLED_ERR_CODE && currencyCode !== FIO_STR) {
```

Comment:
We have a file called `translateError.ts` where we were trying to consolidate all this logic. We should move this stuff into there, and then just call the function to get the string.

Reply:
I just tried to add this to `translateError` but the errors are not a different instance of Error so it would always return the error message if it doesn't match any of the Eos errors. This doesn't allow SendScene to tack on the `transaction_failure_message` prefix. Proper fix requires changes to accountbased to change the Error instance first.